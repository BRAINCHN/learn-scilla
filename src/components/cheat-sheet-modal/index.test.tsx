import React from 'react';
import * as renderer from 'react-test-renderer';
import CheatSheetModal from '.';
import { shallow } from 'enzyme';

const t = (s: string) => s;

describe('Cheat Sheet Modal tests', () => {
  const baseComponent = () => <CheatSheetModal t={t} />;

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent()).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the component', () => {
      const wrapper = shallow(baseComponent());
      const assertion = wrapper.find('[data-testid="cheat-sheet-modal"]').length;
      expect(assertion).toBe(1);
    });
  });
});
