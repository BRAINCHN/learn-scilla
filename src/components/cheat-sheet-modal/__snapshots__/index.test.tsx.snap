// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat Sheet Modal tests basic tests matches the snapshot 1`] = `
.c0 {
  outline: none;
  margin: 0;
  font-family: inherit;
  overflow: visible;
  text-transform: none;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  cursor: pointer;
}

.c0::-webkit-file-upload-button {
  font: inherit;
  -webkit-appearance: button;
}

.c0 [type='button'],
.c0 [type='reset'],
.c0 [type='submit'] {
  -webkit-appearance: button;
}

.c0::-moz-focus-inner,
.c0 [type='button']::-moz-focus-inner,
.c0 [type='reset']::-moz-focus-inner,
.c0 [type='submit']::-moz-focus-inner {
  padding: 0;
  border-style: none;
}

.c0 {
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5;
  border-radius: 0.2rem;
}

.c0 {
  background-color: transparent;
  color: #000000;
  border-color: #000000;
  opacity: 0.85;
}

.c0:hover,
.c0:active,
.c0:focus {
  opacity: 1;
}

<span
  data-testid="cheat-sheet-modal"
>
  <button
    aria-label="Cheat Sheet"
    className="c0"
    data-testid="toggle"
    onClick={[Function]}
    size="small"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 640 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
      />
    </svg>
     
    Cheat Sheet
     
  </button>
</span>
`;
